gson/src/main/java/com/google/gson/stream/JsonReader.java

```diff
@@ -1172,6 +1172,7 @@ private String nextUnquotedValue() throws IOException {
     while (true) {
       for (; pos + i < limit; i++) {
         switch (buffer[pos + i]) {
+          case '"':
```

Comment:
Maybe for consistency it would also be necessary to change `JsonReader#skipUnquotedValue` so that reading a value and skipping a value is consistent, and adding a test for this.

Not sure if `JsonReader#isLiteral` would have to be changed as well, maybe not.